src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala

```diff
@@ -70,35 +68,44 @@ abstract class ErgoNodeViewHolder[StateType <: ErgoState[StateType]](settings: E
     * Restore a local view during a node startup. If no any stored view found
     * (e.g. if it is a first launch of a node) None is to be returned
     */
-  override def restoreState: Option[NodeView] = {
-    ErgoState.readOrGenerate(settings, Some(self)).map { stateIn =>
-      //todo: ensure that history is in certain mode
-      val history = ErgoHistory.readOrGenerate(settings, timeProvider)
-      val wallet = ErgoWallet.readOrGenerate(settings)
-      val memPool = ErgoMemPool.empty
-      val state = restoreConsistentState(stateIn.asInstanceOf[MS], history)
-      (history, state, wallet, memPool)
-    }
+  override def restoreState: Option[NodeView] = if (ErgoHistory.historyDir(settings).listFiles().isEmpty) {
+    None
+  } else {
+    val history = ErgoHistory.readOrGenerate(settings, timeProvider)
+    val wallet = ErgoWallet.readOrGenerate(settings)
+    val memPool = ErgoMemPool.empty
+    val state = restoreConsistentState(ErgoState.readOrGenerate(settings, Some(self)).asInstanceOf[MS], history)
+    Some((history, state, wallet, memPool))
   }
 
-  private def restoreConsistentState(state: StateType, history: ErgoHistory) = {
-    //TODO do we need more than 1 block here?
-    history.bestFullBlockOpt.map { fb =>
-      if (!(state.version sameElements fb.id)) {
-        state.applyModifier(fb) match {
-          case Success(s) =>
-            log.info(s"State and History are inconsistent on startup. Applied missed modifier ${fb.encodedId}")
-            s
-          case Failure(e) =>
-            //TODO catch errors here
-            log.error(s"Failed to apply missed modifier ${fb.encodedId}. Try to resync from genesis", e)
-            ErgoApp.forceStopApplication(500)
-            state
+  private def restoreConsistentState(state: StateType, history: ErgoHistory): StateType = {
+    if (history.bestFullBlockIdOpt.isEmpty) {
+      state
```

Comment:
So if no full blocks in history, state is consistent? What is the logic behind that?